clippy_lints/src/strings.rs

```diff
@@ -399,13 +399,17 @@ impl<'tcx> LateLintPass<'tcx> for StrToString {
             && let ty::Ref(_, ty, ..) = ty.kind()
             && ty.is_str()
         {
-            span_lint_and_help(
+            let mut applicability = Applicability::MachineApplicable;
+            let snippet = snippet_with_applicability(cx, self_arg.span, "..", &mut applicability);
+
+            span_lint_and_sugg(
                 cx,
                 STR_TO_STRING,
                 expr.span,
                 "`to_string()` called on a `&str`",
-                None,
                 "consider using `.to_owned()`",
+                format!("{snippet}.to_owned()"),
```

Comment:
I think having two `.to_owned()` here seems a bit redundant? How about just using `try` as the help msg instead?

```rust
"consider using `.to_owned()`",
format!("{snippet}.to_owned()"),
```

Reply:
(drive-by comment)

It's worth noting that the `rustc-dev-guide` [explicitly state](https://rustc-dev-guide.rust-lang.org/diagnostics.html#suggestion-style-guide) that the suggestion message should "be upfront about what the suggestion is" and explicitly say to avoid "did you mean"/"the following" ...

I would also note that in IDEs, users may not have access to the terminal output of the compiler and may only have access to the help message to decide if they should apply the suggestion and "try" is less than helpful is those cases.

Reply:
> ...the suggestion message should "be upfront about what the suggestion is"...

Well, my bad, I didn't know that 👀.

But in my defense, there are already quite amount of suggestions with "try" existing in clippy. like [`nonminimal_bool`](https://github.com/rust-lang/rust-clippy/blob/4f3180adac9ff2da34c319fe17baa19e2580d09b/clippy_lints/src/booleans.rs#L171-L172), [`manual_swap`](https://github.com/rust-lang/rust-clippy/blob/4f3180adac9ff2da34c319fe17baa19e2580d09b/clippy_lints/src/swap.rs#L160), etc, there are 170+ of them I think, so I though it would make sense to use it.

> ...in IDEs, users may not have access to the terminal output of the compiler and may only have access to the help message to decide if they should apply the suggestion and "try" is less than helpful is those cases.

In that case, maybe clippy could use a "clean up" to replace those existing "try"s (including this one) with more helpful messages? But I don't know if that worth the effort.

Thanks for the heads up~

Reply:
No worries, we also have in `rustc` messages that don't follow our own guidelines, so I wouldn't expect Clippy or other tools to follow it to the letter either.

> In that case, maybe clippy could use a "clean up" to replace those existing "try"s (including this one) with more helpful messages? But I don't know if that worth the effort.

I'm not on the Clippy team so you would have to ask them out that.
Personally I think that could be long term goal.